mock.go

```diff
@@ -79,10 +92,18 @@ func (mo mockMultiOp) Run() error {
 	return nil
 }
 
+func (mo mockMultiOp) RunWithContext(ctx context.Context) error {
+	return mo.WithOptions(Options{Context: ctx}).Run()
+}
+
 func (mo mockMultiOp) RunAtomically() error {
 	return mo.Run()
 }
 
+func (mo mockMultiOp) RunAtomicallyWithContext(_ context.Context) error {
+	return mo.Run()
```

Comment:
Should we be passing the context here so the provided context is definitely used?